jzy3d-core/src/main/java/org/jzy3d/plot3d/rendering/view/Camera.java

```diff
@@ -182,10 +248,33 @@ public float getFar() {
     return far;
   }
 
-  /** Return true if the given point is on the left of the vector eye->target. */
+  /**
+   * Return true if the given (not scaled) point is on the left of the plane build by the
+   * eye->target and up-direction vector, else false.
+   *
+   * @param point not scaled point (its values correspond to a point in the scene)
+   * @return true if the given point is 'on the left side' of this camera
+   */
   public boolean side(Coord3d point) {
-    return 0 < ((point.x - target.x) * (eye.y - target.y)
-        - (point.y - target.y) * (eye.x - target.x));
+    // Compute the sign of the triple product of eye-to-point-direction, up-direction and
+    // eye-to-target-direction (all not scaled). The given point is left of this camera's view-plane
+    // build by eye-to-target-vector and the up-direction, if the following inequation holds
+    // <e-p> * (<up> x <e-t>) > 0
+    if (isOnLeftSide == null) {
+      // Pre-compute and capture the required data in a Predicate to quickly decide (without
+      // creating objects) if a point is on the left of the current camera position.
+
+      Coord3d e = eye.div(scale); // un-scaled eye vector
+      Coord3d viewDirectionU = target.sub(eye).div(scale).normalizeTo(1f);
+      // Left pointing normal vector of the view-plane build by the eye->target-vector and the
+      // up-direction (effectivly the cross product of up-direction and eye-target-vector).
+      // Not scaled but normalized to length one for numerical stability.
+      Coord3d vpn = up.getNormalizedTo(1f).cross(viewDirectionU); // view-plane normal
+
+      isOnLeftSide = p -> 0 < (p.x - e.x) * vpn.x + (p.y - e.y) * vpn.y + (p.z - e.z) * vpn.z;
```

Comment:
I just did not get why you create the function on the fly and record it as a field? I would find more readable to directly execute the code. The fact that the function is cached let me think that lines 267 to 272 will be skipped for the second call to side() hence leading to an old eye/target/scale value.

My point is mainly on the code readability as I tried your code and it really works better than before.

Reply:
The purpose of the Predicate-lambda is to capture the Coord3d objects `e` and `vpn` (as de-facto fields of the Predicate object), in order to avoid re-creating them every time side is called. These Coord3d-objects/vectors do not change while eye/target/up/scale does not change. Since they are captured and the predicate object is re-used (as you already noticed)  runtime is saved since their values are not re-computed and the garbage-collector does not have to clean up a few Coord3d objects for each call to `side()`. So yes, you're right that line 267 to 272 are skipped intentionally when `isOnLeftSide` is `null`.

But `isOnLeftSide` is set to `null` every time `setPosition()` is called (see line 204) in order to trigger a re-computation of the `isOnLeftSide` `Predicate` object. And because each of the fields `eye`, `target`, `up` and `scale` is only set through setPosition() the `isOnLeftSide` never operates on an old value of `eye`, `target`, `up` and `scale`.

In my first commit I created two fields for the values of `e` and `vpn`. But with the intention to have all calculations at one place closely together and to minimize the number of fields I decide to use this approach and tried to make it clear with the comment at the beginning of the if-block.

So if you find this has poor readability I can change this (I think readability is very important). I just saw that the variable `viewDirectionU` is a bit off the schema of the other variables so I would like to fix this anyway.

Reply:
Thanks. Just extended the unit test to reflect this.

I forgot to answer you that you are right about scaling the up vector. Not scaling it is not consistent. However it works because the up vector is simply a vertical direction - changing its length does not change anything for the underlying opengl primitive.